twitter/View Controllers/TimelineViewController.m

```diff
@@ -6,37 +6,138 @@
 //  Copyright © 2018 Emerson Malca. All rights reserved.
 //
 
+#import "AppDelegate.h"
+#import "LoginViewController.h"
 #import "TimelineViewController.h"
 #import "APIManager.h"
+#import "TweetCell.h"
+#import "UIImageView+AFNetworking.h"
+#import "ComposeViewController.h"
+#import "DetailsViewController.h"
+#import "ProfileViewController.h"
+#import "DateTools.h"
+#import "ReplyButton.h"
+#import "ReplyViewController.h"
 
-@interface TimelineViewController ()
+@interface TimelineViewController () <UITableViewDelegate, UITableViewDataSource, ComposeViewControllerDelegate, ReplyViewControllerDelegate, TweetCellDelegate>
+- (IBAction)didTapLogout:(id)sender;
+
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *tweetsArray;
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
 
 @end
 
 @implementation TimelineViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    
+    self.tweetsArray = [[NSMutableArray alloc] init];
+    
+    [self fetchTweets];
     
+    // initialize an UIRefreshControl
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchTweets) forControlEvents:UIControlEventValueChanged];
+    [self.tableView insertSubview:self.refreshControl atIndex:0];
+    
+}
+
+- (void)fetchTweets {
     // Get timeline
     [[APIManager shared] getHomeTimelineWithCompletion:^(NSArray *tweets, NSError *error) {
         if (tweets) {
             NSLog(@"😎😎😎 Successfully loaded home timeline");
-            for (NSDictionary *dictionary in tweets) {
-                NSString *text = dictionary[@"text"];
+            for (Tweet *tweet in tweets) {
+                [self.tweetsArray addObject:tweet];
+                NSString *text = tweet.text;
                 NSLog(@"%@", text);
             }
+            [self.tableView reloadData];
         } else {
             NSLog(@"😫😫😫 Error getting home timeline: %@", error.localizedDescription);
         }
+        [self.refreshControl endRefreshing];
     }];
 }
 
+- (void)fetchNewTimeline {
+    Tweet *lastTweet = self.tweetsArray[self.tweetsArray.count - 1];
```

Comment:
There's a built-in method to get this: lastObject (https://developer.apple.com/documentation/foundation/nsarray/1408316-lastobject?language=objc)

Whether you use lastObject or the count, you should check that the array contains at least one item.